### PR TITLE
Loaders and FetchServiceDetails fix

### PR DESCRIPTION
**Pull request by Vivek205**

### CODE REVIEW

File: src/Redux/actionCreators/ServiceDetailsActions.js [hunk 1]
```diff
@@ -13,8 +14,14 @@ export const UPDATE_FREE_CALLS_REMAINING = "UPDATE_FREE_CALLS_REMAINING";
 const resetServiceDetails = dispatch => {
   dispatch({ type: RESET_SERVICE_DETAILS });
 };
+
+const fetchServiceDetailsFailure = err => dispatch => {
+  dispatch(loaderActions.stopAppLoader);
```
Comment (chetankothari) [inline src/Redux/actionCreators/ServiceDetailsActions.js:18]: You are not using the error message here? Should we show some error message if this fails?

File: src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ const PurchaseDialog = ({ classes, show, onClose, startDepositLoader, startWithd
     try {
       const amountInAGI = amount[txnTypes.DEPOSIT];
       const amountInCogs = agiToCogs(amountInAGI);
-      const response = await sdk.account.depositToEscrowAccount(amountInCogs);
+      await sdk.account.depositToEscrowAccount(amountInCogs);
       setAmount({});
```
Comment (chetankothari) [inline src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js:46]: Why do you need to set the amount here?

File: src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ const PurchaseDialog = ({ classes, show, onClose, startDepositLoader, startWithd
     try {
       const amountInAGI = amount[txnTypes.WITHDRAW];
       const amountInCogs = agiToCogs(amountInAGI);
-      const response = await sdk.account.withdrawFromEscrowAccount(amountInCogs);
+      await sdk.account.withdrawFromEscrowAccount(amountInCogs);
       setAmount({});
```
Comment (chetankothari) [inline src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js:63]: Why do you need to set the amount here?

File: src/components/ServiceDetails/index.js [hunk 4]
```diff
@@ -74,14 +75,20 @@ class ServiceDetails extends Component {
...
+    },
+  } = ownProps;
+
+  return { service: serviceDetails(state, orgId, serviceId), pricing: pricing(state) };
+};
 
 const mapDispatchToProps = dispatch => ({
   fetchServiceDetails: args => dispatch(serviceDetailsActions.fetchServiceDetails({ ...args })),
   resetServiceDetails: () => dispatch(serviceDetailsActions.resetServiceDetails),
+  fetchMeteringData: args => dispatch(serviceDetailsActions.fetchMeteringData({ ...args })),
```
Comment (chetankothari) [inline src/components/ServiceDetails/index.js]: Do you need `resetServiceDetails` and `fetchMeteringData` anymore here?

File: src/components/ServiceDetails/index.js [hunk 5]
```diff
@@ -74,14 +75,20 @@ class ServiceDetails extends Component {
...
+    match: {
+      params: { orgId, serviceId },
+    },
+  } = ownProps;
+
+  return { service: serviceDetails(state, orgId, serviceId), pricing: pricing(state) };
+};
 
 const mapDispatchToProps = dispatch => ({
   fetchServiceDetails: args => dispatch(serviceDetailsActions.fetchServiceDetails({ ...args })),
```
Comment (chetankothari) [inline src/components/ServiceDetails/index.js]: This could be simplified to accept `orgId` and `serviceId`.

**Review by chetankothari — APPROVED** [review]

File: src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js (hunk 2 above)
Comment (Vivek205) [reply src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js:46]: I'm resetting the amount once it is deposited.

File: src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js (hunk 3 above)
Comment (Vivek205) [reply src/components/ServiceDetails/AboutService/ServiceDemo/Purchase/PurchaseDialog/index.js:63]: https://github.com/singnet/refactored-dapp/pull/160#discussion_r316110393

File: src/components/ServiceDetails/index.js (hunk 4 above)
Comment (Vivek205) [reply src/components/ServiceDetails/index.js]: removed https://github.com/singnet/refactored-dapp/pull/160/commits/1c6427e6799311714b153c05dccc87e481d1e1af

**Pull request merged**